ntbs-service/Services/AzureAdDirectoryService.cs

```diff
@@ -141,45 +141,51 @@ private async Task<IEnumerable<Microsoft.Graph.User>> GetAllDirectoryEntries()
             var baseGroups = await _graphServiceClient
             .Groups
             .Request()
-            .Filter($"startswith(displayName, '{_adOptions.BaseUserGroup}')")
+            .Filter($"displayName eq '{_adOptions.BaseUserGroup}'")
             .Select("id, displayName, description")
-            .Top(999)
+            .Top(1)
             .GetAsync();
 
             var baseGroup = baseGroups.FirstOrDefault(g => g.DisplayName == _adOptions.BaseUserGroup);
 
             if (baseGroup != null)
             {
+                var usersInRootGroup = await GetUsersInGroup(baseGroup.Id);
+                listOfDirectoryEntries.AddRange(usersInRootGroup);
 
-                var groupMembers = await _graphServiceClient.Groups[baseGroup.Id]
-                    .Members
-                    .Request()
-                    .GetAsync();
-
-                if (groupMembers != null)
+                var groupMembersRequest = _graphServiceClient.Groups[baseGroup.Id].Members.Request();
+                while (groupMembersRequest != null)
                 {
-                    // get users from root group
-                    var usersInRootGroup = await GetUsersInGroup(baseGroup.Id);
-                    listOfDirectoryEntries.AddRange(usersInRootGroup);
+                    var groupMembers = await groupMembersRequest.GetAsync();
 
-                    foreach (var groupMember in groupMembers)
+                    if (groupMembers != null)
                     {
-                        try
+                        // get users from root group
+                        foreach (var groupMember in groupMembers)
                         {
-                            switch (groupMember.ODataType)
+                            try
                             {
-                                case "#microsoft.graph.group":
-                                    var usersInGroup = await GetUsersInGroup(groupMember.Id);
-                                    listOfDirectoryEntries.AddRange(usersInGroup);
-                                    break;
-                                default:
-                                    break;
+                                switch (groupMember.ODataType)
+                                {
+                                    case "#microsoft.graph.group":
+                                        var usersInGroup = await GetUsersInGroup(groupMember.Id);
+                                        listOfDirectoryEntries.AddRange(usersInGroup);
+                                        break;
+                                    default:
+                                        break;
+                                }
+                            }
+                            catch (Exception)
+                            {
+                                // ignore
```

Comment:
Any idea why we're suppressing exceptions here?

Reply:
I have no idea, and given that they are totally supressed I have no idea how common they are. One approach might be just to remove all of this error suppression, and let Sentry pick them up. If there are somewhat common exceptions it might be nice to see if we can address the root cause instead of just ignoring them. How does that sound?

A more conservative approach would be just to add some logging here, and continue suppressing but at least noting the suppression. Which do you prefer?

Reply:
I've removed them, and dealt with the main error that needed to be supressed (making invalid requests, like trying to get  Group members of objects that aren't groups, like `DirectoryRoles`) as far as I can tell